java/arcs/core/host/AbstractArcHost.kt

```diff
@@ -128,13 +132,16 @@ abstract class AbstractArcHost(vararg initialParticles: ParticleRegistration) :
      * using [ArcHostContextParticle].
      *
      * Subclasses may override this to retrieve the [context] using a different implementation.
+     *
+     * @property arcHostContext a prototype for the final arcHost containing [EntityHandleManager]
      */
-    protected open suspend fun readContextFromStorage(arcId: String): ArcHostContext? =
-        createArcHostContextParticle(arcId).let {
-            it.readArcHostContext(arcId, hostId, this::instantiateParticle)
-        }?.also {
-            contextCache[arcId] = it
-        }
+    protected open suspend fun readContextFromStorage(
+        arcHostContext: ArcHostContext
+    ): ArcHostContext = createArcHostContextParticle(arcHostContext).let {
+        it.readArcHostContext(arcHostContext, hostId, this::instantiateParticle)
+    }?.also {
+        contextCache[arcHostContext.arcId] = it
+    } ?: arcHostContext
```

Comment:
Curious why we want to return the argument if readArcHostContext returned null. Maybe a comment would be good.

Reply:
There used to be a method lookupArcHostContextOrCreate() that would delegate to lookupArcHostContext(), I merged the two. So now it creates a new ArcHostContext() if it fails to find it in the cache, this is an empty, fresh ArcHostContext with an entitymanager, and then tries to deserialize one that might have been previously written to disk. If it succeeds, it returns the new deserialized one with the same entity manager, if it fails to deserialize and return null, then we just return the empty fresh copy (this must be a for a new ArcId that has never been started)

Reply:
added test